dashboard/app/controllers/followers_controller.rb

```diff
@@ -14,6 +14,24 @@ def create
     redirect_to redirect_url, notice: I18n.t('follower.added_teacher', name: @section.teacher.name)
   end
 
+  # As a logged-in student, join a section.  Returns JSON.  Designed to be called by ajax request.
+  def create_async
```

Comment:
@islemaster and @joshlory have been doing a lot of recent thinking about the API for sections, and thus might have opinions on this section